### ISSUE TITLE
Error while using meteo_ogimet() for certain stations in Switzerland

### ISSUE THREAD

**Issue by RPanczak** [bczernecki/climate#58]
Hi team,

I'm using the package to get data for all stations in Switzerland 🇨🇭.  I retrieved all stations in the country using:

```
library(tidyverse)
library(climate)

nearest_stations_ogimet(country = "Switzerland",
                        date = Sys.Date(),
                        add_map = TRUE,
                        point = c(8.2318, 46.7985),
                        no_of_stations = 200
)

ch_stations_ogimet <- nearest_stations_ogimet(country = "Switzerland",
                                              date = Sys.Date(),
                                              point = c(8.2318, 46.7985),
                                              no_of_stations = 200) %>% 
  arrange(wmo_id)
```

I tried to loop through all of them and some seem to throw an error. Bern works fine:

```
# example of one station
bern <- meteo_ogimet(
  date = c(as.Date("2020-02-01"), Sys.Date() - 1),
  # date = c(Sys.Date() - 7, Sys.Date() - 1), 
  interval = "daily",
  coords = FALSE, 
  station = "06631")
```

Whereas here I fail:

```
# example of an error
error <- meteo_ogimet(
  date = c(as.Date("2020-02-01"), Sys.Date() - 1),
  # date = c(Sys.Date() - 7, Sys.Date() - 1), 
  interval = "daily",
  coords = FALSE, 
  station = "06683")

Error in if ((length(test[2, !is.na(test[2, ])]) == 6 & test[2, 5] ==  : 
  argument is of length zero
```

Is there any error in my code? Are the data not available for this station of formatted differently? Or is there a problem with the package? 

My session:

```
R version 4.0.4 (2021-02-15)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 18363)

Matrix products: default

locale:
[1] LC_COLLATE=English_United Kingdom.1252 
[2] LC_CTYPE=English_United Kingdom.1252   
[3] LC_MONETARY=English_United Kingdom.1252
[4] LC_NUMERIC=C                           
[5] LC_TIME=English_United Kingdom.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods  
[7] base     

other attached packages:
 [1] magrittr_2.0.1  forcats_0.5.1   stringr_1.4.0   dplyr_1.0.6    
 [5] purrr_0.3.4     readr_1.4.0     tidyr_1.1.3     tibble_3.1.2   
 [9] tidyverse_1.3.1 ggplot2_3.3.3   climate_1.0.1  

loaded via a namespace (and not attached):
 [1] tidyselect_1.1.1  xfun_0.23         haven_2.4.1      
 [4] colorspace_2.0-1  vctrs_0.3.8       generics_0.1.0   
 [7] htmltools_0.5.1.1 yaml_2.2.1        utf8_1.2.1       
[10] XML_3.99-0.6      rlang_0.4.11      pillar_1.6.1     
[13] glue_1.4.2        withr_2.4.2       DBI_1.1.1        
[16] dbplyr_2.1.1      modelr_0.1.8      readxl_1.3.1     
[19] lifecycle_1.0.0   munsell_0.5.0     gtable_0.3.0     
[22] cellranger_1.1.0  rvest_1.0.0       evaluate_0.14    
[25] labeling_0.4.2    knitr_1.33        curl_4.3.1       
[28] fansi_0.5.0       broom_0.7.6       Rcpp_1.0.6       
[31] scales_1.1.1      backports_1.2.1   jsonlite_1.7.2   
[34] fs_1.5.0          farver_2.1.0      hms_1.1.0        
[37] digest_0.6.27     stringi_1.6.2     grid_4.0.4       
[40] cli_2.5.0         tools_4.0.4       maps_3.3.0       
[43] crayon_1.4.1      pkgconfig_2.0.3   ellipsis_0.3.2   
[46] xml2_1.3.2        data.table_1.14.0 reprex_2.0.0     
[49] lubridate_1.7.10  rstudioapi_0.13   assertthat_0.2.1 
[52] rmarkdown_2.8     httr_1.4.2        R6_2.5.0         
[55] compiler_4.0.4   
```

Thank you very much for help! 🙇

**Comment by aglogowski** [comment]
Dear Radosław,
The station with an error measure only wind speed and this is the specific
exception that wasn't considered in the code:
you can see this below:
Station number 06683
https://www.ogimet.com/cgi-bin/gsynres?lang=en&ind=06683&ndays=32&ano=2020&mes=10&day=01&hora=6&ord=REV&Send=Send
Station number 06631
https://www.ogimet.com/cgi-bin/gsynres?lang=en&ind=06631&ndays=32&ano=2020&mes=10&day=01&hora=6&ord=REV&Send=Send

I will try to fix this issue as quickly as possible.
With regards,
Arkadiusz Głogowski




pon., 31 maj 2021 o 11:41 Radoslaw Panczak ***@***.***>
napisał(a):

**Comment by RPanczak** [comment]
Many thanks Arek!

I wasn't aware that some stations report onl y parital data. This must be the case of the failure in retrieval.

I managed to one by one identify stations that caused the trouble and then getting all the data worked

```
ch_stations_search <- ch_stations_ogimet %>% 
  filter(!wmo_id %in% c("06647", "06656", "06669", "06671", "06683",
                        "06687", "06728", "06730", "06789", "06799"))

```
And then simply

```
ch_weather <- purrr::map_df(ch_stations_search$wmo_id, 
                         meteo_ogimet,
                         date = c(Sys.Date() - 7, Sys.Date() - 1), 
                         interval = "daily",
                         coords = FALSE)
```

It could be probably useful to be able to do it not by trial and error? Is there anyway to get some attribute of station that would allow that perhaps? Or argument to `meteo_ogimet` to skip those stations?

Best,
Radek

**Comment by bczernecki** [comment]
This issue has been partly addressed in climate 1.0.5. Unfortunately, CRAN policy don't allow to pass on failure so I believe the easiest solution would be to use `tryCatch` mechanism. 
Closing.
